packages/ra-core/src/controller/useListParams.ts

```diff
@@ -0,0 +1,282 @@
+import { useCallback, useEffect, useState } from 'react';
+// @ts-ignore
+import { useSelector, useDispatch } from 'react-redux';
+import { parse, stringify } from 'query-string';
+import { push } from 'connected-react-router';
+import lodashDebounce from 'lodash/debounce';
+import isEqual from 'lodash/isEqual';
+import pickBy from 'lodash/pickBy';
+import { Location } from 'history';
+
+import queryReducer, {
+    SET_FILTER,
+    SET_PAGE,
+    SET_PER_PAGE,
+    SET_SORT,
+    SORT_ASC,
+} from '../reducer/admin/resource/list/queryReducer';
+import { crudGetList } from '../actions/dataActions';
+import { changeListParams, ListParams } from '../actions/listActions';
+import { Sort, ReduxState, Identifier, RecordMap } from '../types';
+import removeEmpty from '../util/removeEmpty';
+import removeKey from '../util/removeKey';
+
+interface Props {
+    filter?: object;
+    filterDefaultValues?: object;
+    perPage?: number;
+    sort?: Sort;
+    location: Location;
+    resource: string;
+    debounce?: number;
+}
+
+interface Query extends ListParams {
+    data: RecordMap;
+    filterValues: object;
+    ids: Identifier[];
+    total: number;
+    displayedFilters: {
+        [key: string]: boolean;
+    };
+    requestSignature: any[];
+}
+
+interface Actions {
+    changeParams: (action: any) => void;
+    setPage: (page: number) => void;
+    setPerPage: (pageSize: number) => void;
+    setSort: (sort: Sort) => void;
+    setFilters: (filters: any) => void;
+    hideFilter: (filterName: string) => void;
+    showFilter: (filterName: string, defaultValue: any) => void;
+}
+
+const useListParams = ({
```

Comment:
I'd love a small jsDoc explaining the return value

Reply:
Added some